php/ui/class-component.php

```diff
@@ -0,0 +1,482 @@
+	 */
+	protected function image_icon( $icon ) {
+		$image_atts        = $this->get_attributes( 'icon_image' );
+		$image_atts['src'] = $icon;
+		$html              = array();
+		$html[]            = '<span ' . $this->build_attributes( $this->get_attributes( 'icon' ) ) . ' >';
+		$html[]            = '<img ' . $this->build_attributes( $this->get_attributes( 'icon_image' ) ) . ' />';
+		$html[]            = '</span>';
+
+		return self::compile_html( $html );
+	}
+
+	/**
+	 * Create an icon HTML part, based on the type of icon source.
+	 *
+	 * @return string
+	 */
+	protected function get_icon() {
+
+		$icon   = $this->setting->get_param( 'icon' );
+		$method = 'dashicon';
+		if ( false === strpos( $icon, 'dashicons' ) ) {
+			$method = 'image_icon';
+		}
+
+		return $this->$method( $icon );
+	}
+
+	/**
+	 * Create a Prefix HTML part.
+	 *
+	 * @return string
+	 */
+	protected function prefix() {
+		$html = array(
+			'<span' . $this->build_attributes( $this->get_attributes( 'prefix' ) ) . ' >',
+			$this->setting->get_param( 'prefix' ),
+			'</span>',
+		);
+
+		return self::compile_html( $html );
+	}
+
+	/**
+	 * Create a suffix HTML part.
+	 *
+	 * @return string
+	 */
+	protected function suffix() {
+		$html = array(
+			'<span' . $this->build_attributes( $this->get_attributes( 'suffix' ) ) . ' >',
+			$this->setting->get_param( 'suffix' ),
+			'</span>',
+		);
+
+		return self::compile_html( $html );
+	}
+
+	/**
+	 * Creates the child settings component rendered HTML part.
+	 *
+	 * @return string
+	 */
+	protected function settings() {
+		$html = array();
+		foreach ( $this->setting->get_settings() as $setting ) {
+			$html[] = $setting->render_component();
+		}
+
+		return self::compile_html( $html );
+	}
+
+	/**
+	 * Builds and sanitizes attributes for an HTML tag.
+	 *
+	 * @param array $attributes Array of key value attributes to build.
+	 *
+	 * @return string
+	 */
+	protected function build_attributes( $attributes ) {
+
+		$return = array();
+		foreach ( $attributes as $attribute => $value ) {
+			if ( is_array( $value ) ) {
+				$value = implode( ' ', $value );
+			}
+			$return[] = esc_attr( $attribute ) . '="' . esc_attr( $value ) . '"';
+		}
+
+		return implode( ' ', $return );
+
+	}
+
+	/**
+	 * Compiles HTML parts array into a string.
+	 *
+	 * @param array $html HTML parts array.
+	 *
+	 * @return string
+	 */
+	public static function compile_html( $html ) {
+		return implode( '', $html );
+	}
+
+	/**
+	 * Init the component.
+	 *
+	 * @param Settings\Setting $setting The setting object.
+	 *
+	 * @return self
+	 */
+	public static function init( $setting ) {
+
+		$caller = get_called_class();
+		$type   = $setting->get_param( 'type' );
+		// Check what type this component needs to be.
+		if ( is_callable( $type ) ) {
+			$setting->set_param( 'callback', $setting->get_param( 'type' ) );
+			$setting->set_param( 'type', 'custom' );
+		}
+		// Set Caller.
+		$component = $caller . '\\' . $setting->get_param( 'type' );
```

Comment:
Same. What if we use something like:
`$component = "{$caller}\{$type}";`

Reply:
@pereirinha Yup - changed it.